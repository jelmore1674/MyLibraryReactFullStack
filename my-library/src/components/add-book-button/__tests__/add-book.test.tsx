import AddBookButton from '../add-book';
import { fireEvent, render, screen } from '@testing-library/react';

const mockOnClick = jest.fn(() => 'clicked');

describe('<AddBookButton />', () => {
	it('should render', () => {
		render(<AddBookButton onClick={mockOnClick} classes={''} />);
		expect(screen.getByText('Add Book')).toBeInTheDocument();
	});
	it('should have button role', () => {
		render(<AddBookButton onClick={mockOnClick} classes={''} />);
		expect(screen.getByRole('button')).toBeInTheDocument();
	});
	it('should return clicked when clicked', () => {
		render(<AddBookButton onClick={mockOnClick} classes={''} />);
		fireEvent.click(screen.getByText('Add Book'));
		expect(mockOnClick).toHaveBeenCalled();
	});
	it('should match snapshot', () => {
		const { asFragment } = render(
			<AddBookButton onClick={mockOnClick} classes={''} />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
