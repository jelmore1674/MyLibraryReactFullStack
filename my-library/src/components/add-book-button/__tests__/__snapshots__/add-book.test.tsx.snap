// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBookButton /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-grid"
  >
    <button
      aria-label="button"
      class="btn btn-large btn-success rounded "
    >
      Add Book
    </button>
  </div>
</DocumentFragment>
`;
